megatron/data/blendable_dataset.py

```diff
@@ -34,8 +57,8 @@ def __init__(self, datasets, weights, size, *,
         # Build indicies.
         def _build_indices():
             start_time = time.time()
-            assert num_datasets < 255
-            dataset_index = np.zeros(self.size, dtype=np.uint8)
+            # assert num_datasets < 255
+            dataset_index = np.zeros(self.size, dtype=np.int64)
```

Comment:
I am interested also in overpassing this "num_datasets < 255" limitation.

If I understand correctly this PR, this line is the only thing needed to remove the assertion (?).

I am also wondering:
* In practice, in terms of performance, is it a problem to have thousands of tokenized files? (instead of some dozens)
* To workaround this limitation (of 255), is there a script somewhere to concatenate several ".bin" and ".idx" files (with tokenized text inside)?

Reply:
we have a fix on our fork at 

https://github.com/argonne-lcf/megatron-deepspeed/

if you want to try working off there temporarily. 

i'll work on isolating these changes in a branch today and will send a PR with this fix